dask_ml/decomposition/pca.py

```diff
@@ -223,11 +248,30 @@ def _fit(self, X):
         else:
             lower_limit = 0
 
-        if not (min(n_samples, n_features) >= n_components >= lower_limit):
+        if self.errors not in {"raise", "warn", "ignore"}:
+            err_msg = "errors={} not in ['raise', 'warn', ignore']"
+            raise ValueError(err_msg.format(self.errors))
+        isnan = np.isnan([n_samples, n_features])
+        if isnan.any():
+            msg = (
+                "At least one of [n_samples, n_features]={} is nan. "
+                "No check can be performed to make "
+                "sure n_components is small enough. To continue, either\n\n"
+                "    * pass X.to_dask_array(lengths=True)  # for Dask DataFrame\n"
+                "    * pass errors='warn' and ensure "
+                "n_components <= min(X.shape)\n"
+            )
+            msg = msg.format([n_samples, n_features])
+            if self.errors == "raise":
+                raise ValueError(msg)
+            if self.errors == "warn":
+                warn(msg)
```

Comment:
Under what conditions do we emit this warning, but the overall `.fit` succeeds?

Reply:
When `solver != "auto"` and the input array has unknown length. Here's the relevant test: https://github.com/dask/dask-ml/pull/543/files#diff-7f86330d39ca24bd1ab2bfd063eb893bR767-R776